Make glossary terms use relative links to other glossary terms

jgkim opened the pull request:
Made English, Italian, Japanese, and Russian glossary terms use relative links to other glossary terms.
Fixed the third item of https://github.com/okfn/opendatahandbook/issues/127.

jgkim committed:
Make glossary terms use relative links to other glossary terms

Made English, Italian, Japanese, and Russian glossary terms use relative links to other glossary terms.
Fixed the third item of https://github.com/okfn/opendatahandbook/issues/127.

rufuspollock commented:
This is _fantastic_ and really useful and brief review indicates this looks very good. I will leave @morchickit as Managing Editor to do any additional review and the merge.

jgkim commented:
@rgrp Thanks. :)

rufuspollock commented:
@morchickit shall we merge this - it looks good and it will create changes for other glossaries.

morchickit commented:
Not yet, I'm trying to see that it's not conflicted with the other Spanish
translation. So still looking at this.
On Aug 1, 2015 10:36 AM, "Rufus Pollock" notifications@github.com wrote:

> @morchickit https://github.com/morchickit shall we merge this - it
> looks good and it will create changes for other glossaries.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/okfn/opendatahandbook/pull/172#issuecomment-126890706
> .

jgkim commented:
@morchickit As I said in the description, this pull request has nothing to do with the Spanish translation. I have never changes any file in the Spanish glossary. And as you know I posted comments to the pull request regarding the Spanish glossary. Thank you for your efforts.

morchickit commented:
Yes, I saw it now.

I am still looking over it to see if anything is not good, but so far,
looks great!

Thank you for your patience!

On Sat, Aug 1, 2015 at 3:52 PM, James G. Kim notifications@github.com
wrote:

> @morchickit https://github.com/morchickit As I said in the description,
> this pull request has nothing to do with the Spanish translation. I have
> never changes any file in the Spanish glossary. And as you know I posted
> comments to the pull request regarding the Spanish glossary. Thank you for
> your efforts.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/okfn/opendatahandbook/pull/172#issuecomment-126923835
> .

morchickit commented on glossary/en/terms/internet/index.md:
```diff
@@ -4,4 +4,4 @@ lang: en
 title: Internet
 ---
 
```
@jgkim -This is good! I just notice that we have the same definition to Internet and Information (which is wrong). I'll fix it after I'll merge this pull request.

morchickit commented on glossary/en/terms/open-data/index.md:
```diff
@@ -4,6 +4,6 @@ lang: en
 ---
 
-Data is open if it can be freely accessed, used, modified and shared by anyone for any purpose - subject only, at most, to requirements to provide attribution and/or [share-alike](/glossary/en/terms/share-alike-license). Specifically, open data is defined by the [Open Definition][od] and requires that the data be **A. Legally open**: that is, available under an open (data) license that permits an ...
+Data is open if it can be freely accessed, used, modified and shared by anyone for any purpose - subject only, at most, to requirements to provide attribution and/or [share-alike](../share-alike-license/). Specifically, open data is defined by the [Open Definition][od] and requires that the data be **A. Legally open**: that is, available under an open (data) license that permits anyone freely to  ...
```
@jgkim - I also need to fix the [open definition link]... I'll do it after this pull request, unless you want to fix it before I merge?

jgkim replied:
@morchickit I will fix the `open definition` link before you merge.

morchickit commented on glossary/en/terms/open-development/index.md:
```diff
@@ -4,4 +4,4 @@ lang: en
 ---
 
-Open development seeks to bring the philosophy of the [open movement](http://new.opendatahandbook.org/glossary/en/terms/open-movement/) to international development. It promotes [open government](/glossary/en/terms/open-government/), transparency of aid flows, engagement of beneficiaries in the design and implementation of development projects, and availability and use of open development data.
+Open development seeks to bring the philosophy of the [open movement](http://new.opendatahandbook.org../open-movement/) to international development. It promotes [open government](../open-government/), transparency of aid flows, engagement of beneficiaries in the design and implementation of development projects, and availability and use of open development data.
```
@jgkim can you fix the Open Movement before I merge to match the other links that you did? We should have deleted the http://new.opendatahandbook.org and obiousvly didn't.. sorry!

jgkim replied:
@morchickit I will fix the `open movement` link also.

morchickit commented on glossary/en/terms/open-format/index.md:
```diff
@@ -5,7 +5,7 @@ title: Open format
 <<<<<<< HEAD
-A [file format](/glossary/en/terms/file-format/) whose structure is set out in agreed [standards](/glossary/en/terms/standard), overseen and published by a non-commercial expert body. A file in an open format enjoys the guarantee that it can be correctly read by a range of different software programs or used to pass information between them. Compare [proprietary](/glossary/en/terms/proprietary/).
+A [file format](../file-format/) whose structure is set out in agreed [standards](../standard/), overseen and published by a non-commercial expert body. A file in an open format enjoys the guarantee that it can be correctly read by a range of different software programs or used to pass information between them. Compare [proprietary](../proprietary/).
 =======
```
@jgkim - if I'm not mistaken, this is a conflict? Do you want me to fix it or you?

jgkim replied:
@morchickit I cannot see any conflict here. What do you think should be fixed?

morchickit replied:
This is a conflict in this term, but it's not from you. 
I will push this version and fix it. :-)

morchickit commented on glossary/en/terms/re-use/index.md:
```diff
@@ -5,4 +5,4 @@ title: Re-use
 ---
 
 
-It is rare that data gathered for a particular purpose does not have other possible uses. Happily, data is an infinite resource (see [tragedy of the anti-commons](/glossary/en/terms/tragedy-of-the-anti-commons/)); once gathered, for whatever reason, it can be re-used again and again, in ways that were never envisaged when it was collected, provided only that the data-holder makes it available und ...
```
There is a )) at the end here. @jgkim  - Can you delete one of them? Again, it's a fault in the orginal (probably me), but if we already doing this, it might be good to fix it.

jgkim replied:
@morchickit I think it's not a bug. Actually, the second `)` is matched with the opening parenthesis before `see [tragedy ...`

morchickit commented on glossary/en/terms/resource/index.md:
```diff
@@ -5,4 +5,4 @@ title: Resource
 
 
-CKAN uses this term to denote one of the individual data objects (a file such as a spreadsheet, or an [API](/glossary/en/terms/api/)) in a [dataset](/glossary/en/terms/dataset/).
+CKAN uses this term to denote one of the individual data objects (a file such as a spreadsheet, or an [API](../api/)) in a [dataset](../dataset/).
```
Again, double )) in the api term.

jgkim replied:
@morchickit Here again, the second `)` has its matching opening parenthesis before `a file such`.

morchickit commented on glossary/he/index.md:
```diff
@@ -95,4 +95,4 @@ Information Asset Register רשימות IAR מיועדות לאסוף ולארג
 
 Open Government Data :מונח:’מידע פתוח הנוצר על-ידי הממשלה.’ זהו מידע הנוצר במהלך הפעילות השגרתית של משרדי הממשלה ,מבלי שהוא חושף יחידים או מידע מסחרי רגיש. מידע ממשלתי פתוח הינו תת-מערך של :מונח:’מידע מהסקטור הציבורי’, שהינו רחב יותר. לפרטים, ראו <http://opengovernmentdata.org>.
 
-Tab-separated values ערכים מופרדי טאב (TSV), הם צורה מקובלת של פורמט קבצי טקסט לשיתוף מידע טבלאי. פורמט זה פשוט מאוד ו:מונח:’קריא על-ידי מחשב.
```
I am a bit confused about the change that was done here? In any case, let's leave RTL languages aside for the moment. :-)

jgkim replied:
@morchickit I have never made any change in terms of content. I think the trailing new line character has added to the file, and this should be OK.

morchickit replied:
@jgkim - Hebrew is my Native language, this is why I got confused. Understood now.

morchickit commented on glossary/it/terms/app-application/index.md:
```diff
@@ -3,4 +3,4 @@ section: terms
 title: App / Applicazione
 ---
-Un pezzo di software (abbreviazione per 'app'), disegnato in particolar modo per essere eseguito via [web](/glossary/it/web/) o da un dispositivo mobile o piattaforme analoghe. Una app è in grado di collegarsi verso grandi banche data e quindi essere un potente modo per [consuming](/glossary/it/consuming)/{consumare} dati aperti, che possono essere {real-time}, personalizzati, e (attraverso l'uso ...
+Un pezzo di software (abbreviazione per 'app'), disegnato in particolar modo per essere eseguito via [web](../web/) o da un dispositivo mobile o piattaforme analoghe. Una app è in grado di collegarsi verso grandi banche data e quindi essere un potente modo per [consuming](../consuming/)/{consumare} dati aperti, che possono essere {real-time}, personalizzati, e (attraverso l'uso di un telefono cel ...
```
We need to delete the /{consumare}

jgkim replied:
@morchickit I will delete it.

morchickit commented on glossary/it/terms/data-integration/index.md:
```diff
@@ -4,4 +4,4 @@ lang: it
 ---
 
-Generalmente qualsiasi uso interessante dei dati avviene dalla combinazione con differenti sorgenti. Per permettere questo è necessario garantire che i diversi dataset sia compatibili: questi devono usare gli stessi nomi per gli oggetti, le stesse unità di misura o coordinate ecc... Se la [qualità dei dati](glossary/it/data-quality) è buona allora questo processo di integrazione può essere sempli ...
+Generalmente qualsiasi uso interessante dei dati avviene dalla combinazione con differenti sorgenti. Per permettere questo è necessario garantire che i diversi dataset sia compatibili: questi devono usare gli stessi nomi per gli oggetti, le stesse unità di misura o coordinate ecc... Se la [qualità dei dati](glossary/it/data-quality) è buona allora questo processo di integrazione può essere sempli ...
```
Are we changing data-quality link as well?

jgkim replied:
@morchickit I will change it as well.

morchickit commented on glossary/it/terms/licence/index.md:
```diff
@@ -4,4 +4,4 @@ lang: it
 ---
 
-Uno strumento legale attraverso cui chi detiene un [copyright](/glossary/it/copyright/) può concedere alcuni diritti sull'opera protetta. Dati e contenuti sono aperti se accompagnati da una esplicita licenza conforme alla [Open Definition](/glossary/it/open-definition/). Sono disponibili diverse licenze aperte, fra quelle [Creative Commons](/glossary/en/creative-commons/) c'è la CC-BY che richied ...
+Uno strumento legale attraverso cui chi detiene un [copyright](../copyright/) può concedere alcuni diritti sull'opera protetta. Dati e contenuti sono aperti se accompagnati da una esplicita licenza conforme alla [Open Definition](../open-definition/). Sono disponibili diverse licenze aperte, fra quelle [Creative Commons](../creative-commons/) c'è la CC-BY che richiede la sola [attribuzione](../cr ...
```
Attribution should be (../attribution/)

jgkim replied:
@morchickit I will fix this.

morchickit commented on glossary/it/terms/marginal-cost/index.md:
```diff
@@ -3,4 +3,4 @@ section: terms
 title: Costo marginale
 ---
-Si tratta del costo aggiuntivo sostenuto fornendo una singola copia di una risorsa, ad esempio, dati. Per i dati da aprire secondo la [Open Definition](/glossary/en/open-definition/), essi devono essere caricati non oltre il costo marginale. Qualora i dati sono disponibili per il download su Internet il costo marginale di solito è pari a zero. Ci può essere un piccolo costo marginale in casi ecce ...
+Si tratta del costo aggiuntivo sostenuto fornendo una singola copia di una risorsa, ad esempio, dati. Per i dati da aprire secondo la [Open Definition](../open-definition/), essi devono essere caricati non oltre il costo marginale. Qualora i dati sono disponibili per il download su Internet il costo marginale di solito è pari a zero. Ci può essere un piccolo costo marginale in casi eccezionali. A ...
```
We should delete the {} on the cost recovery.

jgkim replied:
@morchickit I will delete `{}` and the separator `-` on it, but I guess it would be better to completely delete the last sentence `Confronta {cost-recovery}.`.

morchickit commented on glossary/it/terms/open-definition/index.md:
```diff
@@ -4,4 +4,4 @@ lang: it
 ---
 
-La Open Definition ("la definizione di open"), ha avuto il suo primo rilascio da {Open Knowledge} nel 2005 e stabilisce a quali condizioni i dati e i contenuti sono considerati open (aperto). Lo "standard" fornito dalla Open Definition è cruciale perché gran parte del valore degli open data sta nella facilità con cui diverse fonti open data possono essere combinate. La compatibilità giuridica e t ...
+La Open Definition ("la definizione di open"), ha avuto il suo primo rilascio da {Open Knowledge} nel 2005 e stabilisce a quali condizioni i dati e i contenuti sono considerati open (aperto). Lo "standard" fornito dalla Open Definition è cruciale perché gran parte del valore degli open data sta nella facilità con cui diverse fonti open data possono essere combinate. La compatibilità giuridica e t ...
```
We should take the {} off the Open Knowledge

jgkim replied:
@morchickit I will do that.

morchickit commented on glossary/it/terms/open-government/index.md:
```diff
@@ -4,4 +4,4 @@ lang: it
 ---
 
-L'Open Government, e in linea con il movimento {open} in generale, cerca di rendere il funzionamento dei governi [trasparenti](/glossary/en/transparency/), responsabili e sensibili per i cittadini. Ciò include anche gli ideali di democrazia, del giusto processo, della partecipazione dei cittadini e degli open data della pubblica amministrazione. Un approccio completo di Open Government, dovrebbe  ...
+L'Open Government, e in linea con il movimento {open} in generale, cerca di rendere il funzionamento dei governi [trasparenti](../transparency/), responsabili e sensibili per i cittadini. Ciò include anche gli ideali di democrazia, del giusto processo, della partecipazione dei cittadini e degli open data della pubblica amministrazione. Un approccio completo di Open Government, dovrebbe cercare di ...
```
We should take off the {} from the word open

jgkim replied:
@morchickit I will do so.

morchickit commented on glossary/it/terms/open-science/index.md:
```diff
@@ -4,4 +4,4 @@ lang: it
 ---
 
-La pratica della scienza in accordo con i principi {open}, che includono la pubblicazione [open access](/glossary/en/open-access/), la pubblicazione e la collaborazione intorno ai [dati della ricerca](/glossary/en/research-data/) come open data  e il rilascio del [codice sorgente](/glossary/en/source-code/) utilizzato nel crearli, e nell'uso e nello sviluppo di strumenti [open source](/glossary/e ...
+La pratica della scienza in accordo con i principi {open}, che includono la pubblicazione [open access](../open-access/), la pubblicazione e la collaborazione intorno ai [dati della ricerca](../research-data/) come open data  e il rilascio del [codice sorgente](../source-code/) utilizzato nel crearli, e nell'uso e nello sviluppo di strumenti [open source](../open-source/) per l'elaborazione dei d ...
```
Again, { } on the word open.

jgkim replied:
@morchickit I will take off that.

morchickit commented on glossary/it/terms/unconference/index.md:
```diff
@@ -4,4 +4,4 @@ lang: it
 ---
 
-Un meeting, simile ad una conferenza, ma senza un programma stabilito all'inzio. Utilizzando varie tecniche consolidate, i partecipanti concordano congiuntamente il giorno in cui verranno eseguite le sessioni. Alcune sessioni saranno più tradizionali anche includendo relatori ad invito. Un formato popolare fra le comunità tecnologiche è quello di combinare o affiancare una non-conferenza con una  ...
+Un meeting, simile ad una conferenza, ma senza un programma stabilito all'inzio. Utilizzando varie tecniche consolidate, i partecipanti concordano congiuntamente il giorno in cui verranno eseguite le sessioni. Alcune sessioni saranno più tradizionali anche includendo relatori ad invito. Un formato popolare fra le comunità tecnologiche è quello di combinare o affiancare una non-conferenza con una  ...
```
we should delete the ({}) from the phrase community engagement.

jgkim replied:
@morchickit I will delete the whole phrase `({community engagement})`.

morchickit commented:
OK,  @jgkim  - I went over 310 files. There are some bugs, but most of them are mistakes that were dragged from past submissions. If you can change accourding to my suggestion, that will be great. If not, I'll merge and change it myself. 

Great work, thank you!

jgkim committed:
Fix some bugs in the glossary

Fixed some bugs in the glossary according to the comments of https://github.com/okfn/opendatahandbook/pull/172.

jgkim commented:
Whoa, great review @morchickit! I have fixed what you pointed in the comments, and have committed the result. Please, review again. Thank you!

morchickit commented:
OK, now, after merging this, all the links are not working and there is a 404 error. 
@pwalsh , @rgrp ?

rufuspollock commented:
It's working for me. Where are you getting errors?

jgkim commented:
@morchickit The problem is that links from the index file `glossary.html` are also relative to the glossary index itself. We should apply `markdownify` filter to the content of each term, and we also need a hack or something to replace relative links to reflect the current base url. I will look into the problem, but currently I could not find a clear solution yet.

rufuspollock commented:
Just to note i think the main thing would be to remove links in the summary on the terms page (and perhaps even shorten entries and have ellipsis). The way to do this is to htmlize and then stringize. This is something we do with the post summaries on the front page of http://okfnlabs.org/. Should not be hard to dig out the code for this.
